src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -45,8 +45,19 @@ CustomTextLayout::CustomTextLayout(gsl::not_null<IDWriteFactory1*> const factory
     for (const auto& cluster : clusters)
     {
         const auto cols = gsl::narrow<UINT16>(cluster.GetColumns());
+        const auto text = cluster.GetText();
+
+        // Push back the number of columns for this bit of text.
         _textClusterColumns.push_back(cols);
-        _text += cluster.GetText();
+
+        // If there is more than one text character here, push 0s for the rest of the columns
+        // of the text run.
+        for (auto i = 1; i < text.size(); ++i)
+        {
+            _textClusterColumns.push_back(0);
+        }

```

Comment:
```suggestion
        _textClusterColumns.resize(text.size(), 0); // expand vector to text length, but filled with 0s
```

Reply:
alternatively, `std::fill_n(std::back_inserter(_textClusterColumns), count, 0);`

Reply:
I used fill over fill_n just because I had an inkling that the optimizer might be sad about that one too.

Reply:
nnnn fill_n isn't quite right because of the insertion nature of this. trying fill_n and hoping for the best.

Reply:
fill_n with back_inserter works, but now Dustin has another idea of using .resize which sounds even better to me.